pypesto/engine/mpi_pool.py

```diff
@@ -0,0 +1,40 @@
+from mpi4py.futures import MPIPoolExecutor
+from mpi4py import MPI
+import cloudpickle as pickle
+import logging
+from typing import List
+
+from .base import Engine
+from .task import Task
+
+
+logger = logging.getLogger(__name__)
+
+
+def work(pickled_task):
+    """Unpickle and execute task."""
+    task = pickle.loads(pickled_task)
+    return task.execute()
+
+
+class MPIPoolEngine(Engine):
+    """
+    Parallelize the task execution using multiprocessing.
```

Comment:
MPI?

Reply:
maybe even add a link to a good webpage on how mpi works etc.

Reply:
maybe just https://mpi4py.readthedocs.io/en/stable/, anything deeper is probably not that helpful here